hkmc2DiffTests/src/test/scala/hkmc2/MLsDiffMaker.scala

```
@@ -99,7 +99,7 @@ abstract class MLsDiffMaker extends DiffMaker:
       liftDefns = Opt.when(liftDefns.isSet)(LiftDefns()),
       stageCode = stageCode.isSet,
       target = if wasm.isSet then CompilationTarget.Wasm else CompilationTarget.JS,
-      rewriteWhileLoops = !dontRewriteWhile.isSet,
+      rewriteWhileLoops = false //!dontRewriteWhile.isSet,
```

Then, it would make more sense to change the flag to `rewriteWhile` and actually use its value. Of course we don't want to leave a flag that's silently ignored!

---

Fixed in https://github.com/hkust-taco/mlscript/pull/356/commits/43fdc78b728c47211c2eaa437970ff8bf53aefac

---

It looks like the command is never tested.

---

Tests added [here](https://github.com/hkust-taco/mlscript/pull/356/commits/70ed84a84cd079a3a2ad9937cf91818d3eff45d3#diff-b2a55c83b18544e26b05f237542cc7ad4dc9c4850233abecd8ebf00f09d2b202R380-R427)

---

https://github.com/hkust-taco/mlscript/pull/356#discussion_r2591266178